web/src/main/java/com/epam/esm/controller/TagController.java

```diff
@@ -57,12 +73,10 @@ public TagController(TagService tagService) {
         return ResponseEntity.noContent().build();
     }
 
-    @PostMapping(consumes = "application/json")
-    public ResponseEntity<TagDto> createTag(@RequestBody @Validated({PatchGroup.class, UpdateGroup.class}) @Valid TagDto tagDto) {
-        TagDto savedTag = tagService.save(tagDto);
-        return ResponseEntity.ok(savedTag);
-
+    @GetMapping("/max")
+    public ResponseEntity<TagDto> getUserWithMaxOrderCost() {
+        TagDto tagDto = tagService.findMostWidelyUsed();
```

Comment:
You have to 'Get the most widely used tag OF A USER with the highest cost of all orders'.
I don't see any link to a user

Reply:
Implementation of this part of task searches a user with the most order cost in db(this statement is a part of native sql query) and then uses the user to find the most widely used tag. Early i didn't use native query and my expresison consisted of several method including getUserWithMaxOrderCost(), getTagWithMaxOrderCost etc. So i forgot to rename the method i moved to single native query. I renamed the method in commit 617b80b.

Reply:
As you said you search a user with the most order cost in DB, but acceptance criteria says that clients want to get the most widely used tag OF A USER with the highest cost of all orders. Any user, not a user with the most order cost.
For example, a client sends his user id and retrieve the most popular tag exactly for him

Reply:
This method now returns a list of most widely used tags(if there are more than one tag) of user with passed id.
Commit 4bba37e.